artifact-manifest-spec.md

```diff
@@ -0,0 +1,114 @@
+
+- **`mediaType`** *string*
+
+  This field contains the `mediaType` of this document, differentiating from [image-manifest][oci-image-manifest-spec] and [oci-image-index]. The mediaType for this manifest type MUST be `application/vnd.oci.artifact.manifest.v1+json`, where the version WILL change to reflect newer versions. Artifact authors SHOULD support multiple `mediaType` versions to provide the best user experience for their artifact type.
+
+- **`artifactType`** *string*
+
+  A unique value, [defining the type of unique artifact type][artifact-type], as registered with iana.org. See [registering unique types.][registering-iana]. OCI Artifacts that used `oci.image.manifest` used the `manifest.config.mediaType` to differentiate the type of artifact, differentiating from a container image. Artifact authors that implement `oci.artifact.manifest` use `artifactType` to differentiate the type of artifact, freeing up the `config.mediaType` to be OPTIONAL, or shared config schemas between multiple artifact types.
+
+- **`config`** *[descriptor](descriptor.md)*
+
+    This OPTIONAL property references the configuration object for an artifact. When provided, the configuration object MUST be defined with a [descriptor][descriptor], but MAY be any schema defined by the artifact author. As with any descriptor, the `descriptor.mediaType` MUST be registered with iana.org for uniqueness. The `config.mediaType` MAY use any existing schema, and artifact authors are encouraged to utilize existing schemas to simplify artifact implementations.
+
+- **`blobs`** *array of objects*
+
+    A collection of blobs, that represent a collection of 1 or more files. The blobs array is analogous to [oci.image.manifest layers][oci-image-manifest-spec-layers], however unlike [image-manifest][oci-image-manifest-spec], the ordering of blobs is specific to the artifact type. Some artifacts may choose an overlay of files, while other artifact types may store collections of files in different locations. To reflect the artifact specific implementations, the `oci.image.manifest.layers` collection is renamed to `blobs`.
+
+    Each item in the array MUST be a [descriptor][descriptor].
+
+    An encountered `mediaType` that is unknown to the implementation MUST be ignored.
+
+- **`references`** *array of objects*
+
+   An OPTIONAL collection of referenced artifacts. When used, the artifact defined within this manifest is said to be dependent upon the referenced `manifests`.
+
+   As manifests within the `[references]` array are deleted, this manifest MAY be deleted. [Distribution spec][oci-distribution-spec] implementations that implement de-duplication, MAY implement ref counting.
```

Comment:
Are there any current conversations about deleting a manifest vs a blob? AIUI, if you wanted to delete a blob you would have to delete the whole manifest. That should solve the signature and SBoM dependency issue.

Reply:
can you elaborate on what your thinking?

Generally speaking, a blob is an internal detail of an artifact. You shouldn’t delete a blob, rather you delete the artifact (which is represented as a manifest)

When you delete an artifact, the registry does ref counting and reconciles whether it should delete the blobs. 

For SBoMs, signatures, Nydus image layers and other extension artifacts, we provide linking manifests so that we can track the artifacts through their manifest.

Reply:
"As manifests within the references are deleted this manifest MAY be deleted"
To me this indicates that a manifest can be deleted if it has no references. Can it be deleted if it has references?

Reply:
I think the important word there is "this" manifest. If A has references to B, and B is deleted, then A may be deleted by a GC. There's no requirement that a registry provide a GC, but users cannot assume A will continue to exist. Explicitly deleting either A or B should always be possible. To give an example, you can remove a signature, and you can remove the manifest the signature points to, and if you remove the manifest, the signature could get deleted automatically by the registry.

Reply:
resolving as I believe we've covered